src/main/java/duke/DukeCommandMatcher.java

```diff
@@ -0,0 +1,167 @@
+package duke;
+
+import duke.exceptions.*;
+import duke.storage.Storage;
+import duke.tasks.*;
+import duke.utils.Constants;
+import duke.utils.UtilFunction;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+
+public class DukeCommandMatcher {
+
+    private SingletonTaskList taskList;
+
+    private static final List<String> commandList = new ArrayList<>(Arrays.asList(Constants.LISTPATTERN,
+            Constants.EXITPATTERN, Constants.DONEPATTERN, Constants.TODOPATTERN, Constants.DEADLINEPATTERN,
+            Constants.EVENTPATTERN, Constants.DELETEPATTERN, Constants.FINDPATTERN
+            ));
+
+    public DukeCommandMatcher(Storage database) {
+        this.taskList = SingletonTaskList.getInstance(database);
+    }
+
+
+    public String matchCommand(String command) throws CommandNotFoundException, NullCommandException,
```

Comment:
Yeah, it does not look nice, but because of the same behaviour that Duke would respond to all these exceptions, I decide to throw all of them and handle it together in the highest level. Thanks for the suggestion.